.generator/src/generator/templates/api.j2

```diff
@@ -0,0 +1,223 @@
+        fail ArgumentError, 'invalid value for "{% if parameter.required %}{{ name|attribute_name }}{% else %}opts[:"{{ name|attribute_name }}"]{% endif %}" when calling {{classname}}.{{ operation.operationId|snake_case }}, must be greater than {% if not parameter.exclusiveMinimum %}or equal to {% endif %}{{ schema.minimum }}.'
+      end
+
+  {%- endif %}
+  {%- if schema.pattern is defined %}
+      pattern = Regexp.new({{ schema.pattern }})
+      if @api_client.config.client_side_validation && {% if not parameter.required %}!opts[:'{{ name|attribute_name }}'].nil? && {% endif %}{% if parameter.required %}{{ name|attribute_name }}{% else %}opts[:'{{ name|attribute_name }}']{% endif %} !~ pattern
+        fail ArgumentError, "invalid value for '{% if parameter.required %}{{ name|attribute_name }}{% else %}opts[:\"{{ name|attribute_name }}\"]{% endif %}' when calling {{classname}}.{{ operation.operationId|snake_case }}, must conform to the pattern #{pattern}."
+      end
+
+  {%- endif %}
+  {%- if schema.maxItems is defined %}
+      if @api_client.config.client_side_validation && {% if not parameter.required %}!opts[:'{{ name|attribute_name }}'].nil? && {% endif %}{% if parameter.required %}{{ name|attribute_name }}{% else %}opts[:'{{ name|attribute_name }}']{% endif %}.length > {{ schema.maxItems }}
+        fail ArgumentError, 'invalid value for "{% if parameter.required %}{{ name|attribute_name }}{% else %}opts[:"{{ name|attribute_name }}"]{% endif %}" when calling {{classname}}.{{ operation.operationId|snake_case }}, number of items must be less than or equal to {{ schema.maxItems }}.'
+      end
+
+      {%- endif %}
+  {%- if schema.minItems is defined %}
+      if @api_client.config.client_side_validation && {% if not parameter.required %}!opts[:'{{ name|attribute_name }}'].nil? && {% endif %}{% if parameter.required %}{{ name|attribute_name }}{% else %}opts[:'{{ name|attribute_name }}']{% endif %}.length < {{ schema.minItems }}
+        fail ArgumentError, 'invalid value for "{% if parameter.required %}{{ name|attribute_name }}{% else %}opts[:"{{ name|attribute_name }}"]{% endif %}" when calling {{classname}}.{{ operation.operationId|snake_case }}, number of items must be greater than or equal to {{ schema.minItems }}.'
+      end
+
+  {%- endif %}
+{%- endfor %}
+      # resource path
+      local_var_path = '{{ path }}'{% for parameter in operation.parameters if parameter.in == "path" %}.sub('{{ '{' }}{{ parameter.name }}{{ '}'}}', CGI.escape({{ parameter.name }}.to_s).gsub('%2F', '/')){% endfor %}
+
+      # query parameters
+      query_params = opts[:query_params] || {}
+      {%- for name, parameter in operation|parameters if parameter.in == "query" and parameter.required %}
+      {%- set collection = parameter|parameter_schema|collection_format %}
+      query_params[:'{{ name }}'] = {% if collection %}@api_client.build_collection_param({{ name|attribute_name }}, :{{ collection }}){% else %}{{ name|attribute_name }}{% endif %}
```

Comment:
Something is wrong around that logic, not sure what.

Reply:
solved